code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -93,6 +93,14 @@ func (sp *stakePool) save(sscKey, blobberID string,
 	r, err := balances.InsertTrieNode(stakePoolKey(sscKey, blobberID), sp)
 	logging.Logger.Debug("after stake pool save", zap.String("root", r))
 
+	data := dbs.DbUpdates{
+		Id: blobberID,
+		Updates: map[string]interface{}{
+			"offers_total": int64(sp.TotalOffers),
+		},
+	}
+	balances.EmitEvent(event.TypeStats, event.TagUpdateBlobber, blobberID, data)
```

Comment:
A lot of the methods that call stalePool.save immediately emit an `event.TagUpdateBlobber` event. Although it's more code, it might be more efficient to move  `emitting the TotalOffers` up the stack.

Reply:
there is a problem with such approach, if we move it up the stack we will copy-past event emit all over the code, and if you will forget to emit it somewhere data will become incorrect. in general it is a big flaw of current implementation, we need to make it more centralized